lib/msf/core/exploit/file_dropper.rb

```diff
@@ -40,7 +40,7 @@ def initialize(info = {})
   #   exploiting).
   # @return [void]
   def register_files_for_cleanup(*files)
-    @dropped_files += files
+    @dropped_files += files.map(&:clone)
```

Comment:
I think if we swap this to `dup` it'd be good to go

Reply:
Makes sense to me 👍

Reply:
Absolutely, sounds good to me! I'll make the changes. Thanks!

Reply:
Addressed in [5f438f7](https://github.com/rapid7/metasploit-framework/pull/18403/commits/5f438f729d12db0c334169909e929fd59a509be9)